File: output/llama_4_maverick/codenet/direct_translation/java/python/s831281137.py

import sys

def main():
    N = nint()
    W = nint()
    v = [0] * (N + 1)
    w = [0] * (N + 1)
    for i in range(1, N + 1):
        v[i] = nint()
        w[i] = nint()
    sumV = [[0] * (W + 1) for _ in range(N + 1)]
    for i in range(1, N + 1):
        for i_sum in range(w[i]):
            sumV[i][i_sum] = sumV[i - 1][i_sum]
        for i_sum in range(w[i], W + 1):
            sumV[i][i_sum] = max(sumV[i - 1][i_sum], sumV[i - 1][i_sum - w[i]] + v[i])
    print(sumV[N][W])

def nint():
    return int(sys.stdin.readline())

def nlong():
    return int(sys.stdin.readline())

def nstr():
    return sys.stdin.readline().strip()

def nsToChars():
    return list(sys.stdin.readline().strip())

def nints(n, padL=0, padR=0):
    a = [0] * (padL + n + padR)
    it = iter(map(int, sys.stdin.readline().split()))
    for i in range(n):
        a[padL + i] = next(it)
    return a

def nlongs(n, padL=0, padR=0):
    a = [0] * (padL + n + padR)
    it = iter(map(int, sys.stdin.readline().split()))
    for i in range(n):
        a[padL + i] = next(it)
    return a

def nstrs(n):
    return [sys.stdin.readline().strip() for _ in range(n)]

def nsToChars2D(h, w, pad=0):
    a2 = [[''] * (w + pad * 2) for _ in range(h + pad * 2)]
    for i in range(h):
        a = list(sys.stdin.readline().strip())
        for j in range(w):
            a2[pad + i][pad + j] = a[j]
    return a2

if __name__ == "__main__":
    main()

File: dataset/codenet/Java/Code/s831281137.java
import java.io.PrintStream;
import java.util.Scanner;

public class Main {
    static final Scanner     sc = new Scanner(System.in);
    static final PrintStream so = System.out;

    public static void main(String[] args) {
        int N = nint();
        int W = nint();
        
        // 品物i
        int[] v = new int[N+1];
        int[] w = new int[N+1];
        
        for (int i = 1; i <= N; i++) {
            v[i] = nint();
            w[i] = nint();
        }
        
        int[][] sumV = new int[N+1][W+1];
        
        for (int i = 1; i <= N; i++) {
            for (int i_sum = 0; i_sum < w[i]; i_sum++) {
                sumV[i][i_sum] = sumV[i-1][i_sum];
            }
            
            for (int i_sum = w[i]; i_sum <= W; i_sum++) {
                sumV[i][i_sum] = Math.max(sumV[i-1][i_sum], sumV[i-1][i_sum-w[i]] + v[i]);
            }
        }
        
        so.println(sumV[N][W]);
    }

    private static long nlong() {
        return sc.nextLong();
    }

    private static int nint() {
        return sc.nextInt();
    }

    private static String nstr() {
        return sc.next();
    }

    private static char[] nsToChars() {
        return sc.next().toCharArray();
    }

    private static long[] nlongs(int n) {
        return nlongs(n, 0, 0);
    }

    private static int[] nints(int n) {
        return nints(n, 0, 0);
    }

    private static int[] nints(int n, int padL, int padR) {
        int[] a = new int[padL + n + padR];
        for (int i = 0; i < n; i++)
            a[padL + i] = nint();
        return a;
    }

    private static long[] nlongs(int n, int padL, int padR) {
        long[] a = new long[padL + n + padR];
        for (int i = 0; i < n; i++)
            a[padL + i] = nlong();
        return a;
    }

    private static String[] nstrs(int n) {
        String[] a = new String[n];
        for (int i = 0; i < n; i++)
            a[i] = nstr();
        return a;
    }

    private static char[][] nsToChars2D(int h, int w) {
        return nsToChars2D(h, w, 0);
    }

    private static char[][] nsToChars2D(int h, int w, int pad) {
        char[][] a2 = new char[h + pad * 2][w + pad * 2];
        for (int i = 0; i < h; i++)
            System.arraycopy(nsToChars(), 0, a2[pad + i], pad, w);
        return a2;
    }
}

